aggregates/src/commonMain/kotlin/org/dhis2/mobile/aggregates/model/mapper/TableGroupToTableModel.kt

```diff
@@ -64,15 +62,17 @@ internal suspend fun TableGroup.toTableModel(
     )
 
     val rowHeadersCombinations: List<List<CellElement>> = when (pivotMode) {
-        is PivoteMode.CategoryToColumn ->
-            cellElements.map { cellElement ->
-                buildList {
-                    add(cellElement)
-                    pivotMode.pivotedHeaders.forEach { pivotedHeader ->
-                        add(pivotedHeader)
-                    }
+        is PivoteMode.CategoryToColumn -> {
```

Comment:
do you think that this mapper could be tested for us to prevent future miss functions?